content/docs/examples/advanced-gateways/egress-mongo/index.md

```diff
@@ -0,0 +1,1111 @@
+$ kubectl delete destinationrule egressgateway-for-mongo
+{{< /text >}}
+
+## Egress control for TLS
+
+In the real life, most of the communication to the external services must be encrypted and
+[the MongoDB protocol runs on top of TLS](https://docs.mongodb.com/manual/tutorial/configure-ssl/).
+Also, the TLS clients usually send
+[Server Name Indication](https://en.wikipedia.org/wiki/Server_Name_Indication), SNI, as part of their handshake. If your
+MongoDB server runs TLS and your MongoDB client sends SNI as part of the handshake, you can control your MongoDB egress
+traffic as any other TLS-with-SNI traffic. With TLS and SNI, you do not need to specify the IP addresses of your MongoDB
+servers. You specify their host names instead, which is more convenient since you do not have to rely on the stability of
+the IP addresses. You can also specify wildcards as a prefix of the host names, for example allowing access to any
+server from the `*.com` domain.
+
+To check if your MongoDB server supports TLS, run:
+
+{{< text bash >}}
+$ openssl s_client -connect $MONGODB_HOST:$MONGODB_PORT -servername $MONGODB_HOST
+{{< /text >}}
+
+If the command above prints a certificate returned by the server, the server supports TLS. If not, you have to control
+your MongoDB egress traffic on the TCP level, as described in the previous sections.
+
+### Control TLS egress traffic without a gateway
+
+In case you [do not need an egress gateway](/docs/examples/advanced-gateways/egress-gateway/#use-case), follow the
+instructions in this section. If you want to direct your traffic through an egress gateway, proceed to
+[Direct TCP Egress traffic through an egress gateway](#direct-tcp-egress-traffic-through-an-egress-gateway).
+
+1.  Create a `ServiceEntry` and a `VirtualService` for the MongoDB service:
+
+    {{< text bash >}}
+    $ kubectl apply -f - <<EOF
+    apiVersion: networking.istio.io/v1alpha3
+    kind: ServiceEntry
+    metadata:
+      name: mongo
+    spec:
+      hosts:
+      - $MONGODB_HOST
+      ports:
+      - number: $MONGODB_PORT
+        name: tls
+        protocol: TLS
+      resolution: DNS
+    ---
+    apiVersion: networking.istio.io/v1alpha3
+    kind: VirtualService
+    metadata:
+      name: mongo
+    spec:
+      hosts:
+      - $MONGODB_HOST
+      tls:
+      - match:
+        - port: $MONGODB_PORT
+          sni_hosts:
+          - $MONGODB_HOST
+        route:
+        - destination:
+            host: $MONGODB_HOST
+            port:
+              number: $MONGODB_PORT
+          weight: 100
+      location: MESH_EXTERNAL
+    EOF
+    {{< /text >}}
+
+1.  Refresh the web page of the application. The application should display the ratings without error.
+
+#### Cleanup of the egress configuration for TLS
+
+{{< text bash >}}
+$ kubectl delete serviceentry mongo
+$ kubectl delete virtualservice mongo
+{{< /text >}}
+
+### Direct TLS Egress traffic through an egress gateway
```

Comment:
This would also be good to show how to do it using the new external network endpoint ServiceEntry approach

Reply:
@frankbu Replied here https://github.com/istio/istio.io/pull/2347#discussion_r233744203.